cloud-chatops/src/find_prs.py

```diff
@@ -0,0 +1,74 @@
+"""This module finds all open pull requests from given repositories."""
+
+from typing import List, Dict, Union, Tuple
+import requests
+from read_data import get_token
+from pr_dataclass import PR, PRProps
+
+
+class FindPRs:
+    """This class finds all open pull requests in the given repositories. It can also sort them chronologically."""
+
+    def __init__(self, github_token=""):
+        """
+        Initialise the class with a GitHub token if not using run as entry point.
+        :param github_token: GitHub REST API token.
+        """
+        self.github_token = github_token
+
+    def run(
+        self, repos: Dict[str, List], sort: Union[Tuple[PRProps, bool], None] = None
+    ) -> List[PR]:
+        """
+        Finds all open pull requests and returns them as a list of dataclasses.
+        :param repos: Dictionary of repository names and owners.
+        :param sort: Sort the list by property in ascending or descending order.
+        :return: List of PRs
+        """
+        if not self.github_token:
+            self.github_token = get_token("GITHUB_TOKEN")
+
+        raw_responses = []
+        for organisation in repos:
+            raw_responses += self.request_all_repos(
+                organisation, repos.get(organisation)
+            )
+
+        dataclass_list = [PR.from_json(response) for response in raw_responses]
+
+        if sort:
+            dataclass_list = sorted(
+                dataclass_list,
+                key=lambda pr: getattr(pr, str(sort[0]).split(sep=".")[1].lower()),
+                reverse=sort[1],
+            )
```

Comment:
This shouldn't sort it's doing too many things at the moment, as this class finds PRs AND sorts PRs

This should really be extracted into it's own method
Additionally, we're doing this complex thing:

```
getattr(pr, str(sort[0]).split(sep=".")[1].lower()
```

When it would be much easier to have

```
def sort_by_creation_time(results: List[PR]) -> List[PR]):
   return sorted(results, lambda: pr: pr.created_at)

def sort_by_update_time(results: List[PR]) -> List[PR]):
   return sorted(results, lambda: pr: pr.updated_at)
```

Since I suspect we've got a massive amount of DRY going on for 2 cases, updated and created....